maven-model-builder/src/main/java/org/apache/maven/model/profile/DefaultProfileSelector.java

```diff
@@ -108,6 +168,40 @@ public List<Profile> getActiveProfiles(
         return activeProfiles;
     }
 
+    private Map<String, Profile> earlyInterpolateProfileActivations(
+            Collection<Profile> original, ProfileActivationContext context) {
+
+        org.apache.maven.api.model.Model model = org.apache.maven.api.model.Model.newBuilder()
+                .profiles(original.stream()
+                        .map(p -> org.apache.maven.api.model.Profile.newBuilder()
+                                .id(p.getId())
+                                .activation(Optional.ofNullable(p.getActivation())
+                                        .map(Activation::getDelegate)
+                                        .orElse(null))
+                                .build())
+                        .collect(Collectors.toList()))
+                .build();
+
+        ModelBuildingRequest mbr = new DefaultModelBuildingRequest()
+                .setActiveProfileIds(context.getActiveProfileIds())
+                .setInactiveProfileIds(context.getInactiveProfileIds())
+                .setSystemProperties(asProperties(context.getSystemProperties()))
+                .setUserProperties(asProperties(context.getUserProperties()))
+                .setTwoPhaseBuilding(true)
+                .setValidationLevel(ModelBuildingRequest.VALIDATION_LEVEL_MINIMAL);
+
+        model = interpolator.interpolateModel(
```

Comment:
I wonder if using directly the `MavenTransformer` class would be easier instead of recreating a `ModelBuildingRequest` here....